gpkit/keydict.py

```diff
@@ -131,6 +140,15 @@ def __getitem__(self, key):
     def __setitem__(self, key, value):
         "Overloads __setitem__ and []= to work with all keys"
         key, idx = self.parse_and_index(key)
+        if hasattr(value, "exp") and not value.exp:
+            value = value.value  # substitute constant monomials
```

Comment:
Yeah, there's this awkward overlap between what's required for a variable value and for an array that might contain variable values. I tried representing that more directly, but this was more readable.